bin/delta.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+### Wrapper to calculate deltas between two commits
+
+
+HELP_STRING="
+Check if git branch or commit <NEW> generates any new build output running
+<COMMAND> based on git branch or commit <OLD>.
+
+check-warn.sh [OPTIONS]
+
+DIFF OPTIONS:
+  -a OLD             git branch or commit ID OLD as the base, 'HEAD' is
+                     acceptable
+  -b NEW             git branch or commit ID NEW for checking WARNING(s), 'HEAD'
+                     is acceptable
+  -m, --merge        automatically pick the merge-base for the given commits
+  --                 everything after this is the COMMAND to be executed
+
+OTHER OPTIONS:
+  -h, --help         this message
+"
+
+trap restore_branch 1 2 3 6 15
+
+usage()
+{
+	echo "${HELP_STRING}"
+	exit "$1"
+}
+
+save_branch()
+{
+	# get current branch name or commit ID
+	_cbr=$(git branch | sed -n '/\* /s///p')
+	if [[ "$_cbr" == "(HEAD detached"* ]]; then
+		_cbr=$(git rev-parse HEAD)
+	fi
+}
+
+restore_branch()
+{
+	[ -z "$_cbr" ] || git checkout "$_cbr"
+}
+
+summary()
+{
+	local _num
+
+	rm -f _new-warn.log
```

Comment:
for future use - generate the new and old file names with mktemp? then we could, in theory, if needed run multiple checks simultaneously.